applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -61,39 +60,32 @@ void  ElasticIsotropic3D::CalculateMaterialResponsePK2(ConstitutiveLaw::Paramete
     //b.- Get Values to compute the constitutive law:
     Flags &Options=rValues.GetOptions();
 
-    const Properties& MaterialProperties  = rValues.GetMaterialProperties();
     Vector& StrainVector                  = rValues.GetStrainVector();
     Vector& StressVector                  = rValues.GetStressVector();
-    const double& E          = MaterialProperties[YOUNG_MODULUS];
-    const double& NU    = MaterialProperties[POISSON_RATIO];
 
     //NOTE: SINCE THE ELEMENT IS IN SMALL STRAINS WE CAN USE ANY STRAIN MEASURE. HERE EMPLOYING THE CAUCHY_GREEN
     if(Options.IsNot( ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN ))
     {
+        KRATOS_DEBUG_ERROR_IF_NOT(rValues.IsSetDeformationGradientF()) << "Element does not provide strain, hence F has to be provide. This is not the case" << std::endl;
         CalculateCauchyGreenStrain(rValues, StrainVector);
     }
 
     if( Options.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR ) )
     {
         Matrix& ConstitutiveMatrix = rValues.GetConstitutiveMatrix();
-        CalculateElasticMatrix( ConstitutiveMatrix, E, NU );
+        CalculateElasticMatrix( ConstitutiveMatrix, rValues);
     }
 
     if( Options.Is( ConstitutiveLaw::COMPUTE_STRESS ) )
     {
-        if (rValues.IsSetDeformationGradientF() == true)
```

Comment:
why do you remove this?

Reply:
Because it is computed line 73

Reply:
Actually, this line is needed to the shell elements